pkg/server/event_handler_test.go

```diff
@@ -97,45 +98,87 @@ func TestEventHandler_HttpHandler(t *testing.T) {
 	}{
 		{
 			name: "success",
-			pubsubMessageBytes: []byte(`
+			pubsubMessageBytes: []byte(fmt.Sprintf(`
 			{
-				"message": {
-					"attributes": {
-						"bucketId": "foo",
-						"objectId": "bar"
-					}
+			  "message": {
+				"attributes": {
+				  "bucketId": "foo",
+				  "objectId": "bar",
+				  "payloadFormat": "JSON_API_V1"
 				},
-				"subscription": "test_subscription"
+				"data" : %q
+			  },
+			  "subscription": "test_subscription"
 			}
-			`),
+			`, base64.StdEncoding.EncodeToString([]byte(`{
+				"metadata": {
+				  "git-commit": "test-github-commit",
+				  "triggered-timestamp": "2023-04-25T17:44:57",
+				  "git-workflow-sha": "test-workflow-sha",
+				  "git-workflow": "test-workflow",
+				  "git-repo": "test-github-repo"
+				}
+			  }`)))),
 			gcsObjectBytes: []byte(`foo: bar
 isOK: true`),
 			wantStatusCode:     http.StatusCreated,
 			wantRespBodySubstr: "OK",
 		},
 		{
 			name:               "invalid_request_body",
-			pubsubMessageBytes: []byte(`}`),
+			pubsubMessageBytes: []byte(`}"`),
 			gcsObjectBytes:     nil,
 			wantStatusCode:     http.StatusBadRequest,
 			wantRespBodySubstr: "invalid character",
 		},
 		{
 			name: "failed_handle_event",
-			pubsubMessageBytes: []byte(`
+			pubsubMessageBytes: []byte(fmt.Sprintf(`
 			{
-				"message": {
-					"attributes": {
-						"bucketId": "foo",
-						"objectId": "bar2"
-					}
+			  "message": {
+				"attributes": {
+				  "bucketId": "foo",
+				  "objectId": "bar2",
+				  "payloadFormat": "JSON_API_V1"
 				},
-				"subscription": "test_subscription"
+				"data" : %q
+			  },
+			  "subscription": "test_subscription"
 			}
-			`),
+			`, base64.StdEncoding.EncodeToString([]byte(`{
+				"metadata": {
+				  "git-commit": "test-github-commit",
+				  "triggered-timestamp": "2023-04-25T17:44:57",
+				  "git-workflow-sha": "test-workflow-sha",
+				  "git-workflow": "test-workflow",
+				  "git-repo": "test-github-repo"
+				}
+			  }`)))),
 			wantStatusCode:     http.StatusInternalServerError,
 			wantRespBodySubstr: "failed to get GCS object",
 		},
+		{
+			name: "invalid_pubsubmessage_data",
+			pubsubMessageBytes: []byte(fmt.Sprintf(`
+			{
+			  "message": {
+				"attributes": {
+				  "bucketId": "foo",
+				  "objectId": "bar",
+				  "payloadFormat": "JSON_API_V1"
+				},
+				"data" : %q
+			  },
+			  "subscription": "test_subscription"
+			}
+			`, base64.StdEncoding.EncodeToString([]byte(`{
+				"metadata": {
+					""
+				}
+			  }`)))),
```

Comment:
You can direct replace %q with a random string. It'd be more obvious it's invalid.

Reply:
data's value needs to be of type `[]byte`, putting a string there will cause problem for handler to parse the response body into `PubsubMessage`. Use %q here is just to bypass that.

Reply:
@yolocs `[]byte` is base64 encoded in JSON.

@sailorlqh it would be slightly better to use a real structure here instead of compiling a string. If you need to compile a string, using "%s" would be less indirect than %q.

Reply:
Updated it to a real structure.

Reply:
Oh right I missed that.